src/vs/workbench/contrib/files/common/files.ts

```diff
@@ -164,11 +164,11 @@ export class TextFileContentProvider extends Disposable implements ITextModelCon
 	}
 
 	private static resourceToTextFile(scheme: string, resource: URI): URI {
-		return resource.with({ scheme, query: JSON.stringify({ scheme: resource.scheme }) });
+		return resource.with({ scheme, query: JSON.stringify({ scheme: resource.scheme, query: resource.query }) });
 	}
 
 	private static textFileToResource(resource: URI): URI {
-		return resource.with({ scheme: JSON.parse(resource.query)['scheme'], query: null });
+		return resource.with({ scheme: JSON.parse(resource.query)['scheme'], query: JSON.parse(resource.query)['query'] });
```

Comment:
Instead of calling `JSON.parse(resource.query)` twice, can this be extracted to a variable?

Reply:
I have pushed a change.

Reply:
Actually, 3 changes, each simpler that the previous. Let me know if I have simplified to the point of obscurity and you'd like me to row back.